### PR TITLE
Git difference

### PR DESCRIPTION
**Pull request by sergey-shandar**
Avocado can detect difference between `source` and `target` branch and report only new errors.

### CODE REVIEW

**Comment by vladbarosan** [comment]
adding @billytrend as he implemented something similar for the hub

File: src/index.ts [hunk 1]
```diff
@@ -60,24 +25,94 @@ export type NotAutoRestMarkDown = {
...
-  fs
-    .recursiveReaddir(path.resolve(dir))
-    .filter(f => path.basename(f).toLowerCase() === "readme.md")
-    .flatMap(validateReadMeFile)
+export const avocado = ({ cwd, env }: cli.Config): asyncIt.AsyncIterableEx<Error> =>
+  asyncIt.iterable<Error>(async function*() {
+    const targetBranch = env.SYSTEM_PULLREQUEST_TARGETBRANCH
+    // detect Azure DevOps Pull Request validation.
+    if (targetBranch !== undefined) {
+      const sourceGitRepository = git.repository(cwd)
```
Comment (sergey-shandar) [inline src/index.ts]: 1. `azureDevOps`.
2. `diff` - compare two folders.
2. default.

File: src/index.ts [hunk 2]
```diff
@@ -60,24 +25,94 @@ export type NotAutoRestMarkDown = {
...
+      await sourceGitRepository({ branch: [sourceBranch] })
+      await sourceGitRepository({ branch: [targetBranch, `remotes/origin/${targetBranch}`] })
+      await sourceGitRepository({
```
Comment (sergey-shandar) [inline src/index.ts]: Remove this line (no need for diff)

File: src/index.ts [hunk 3]
```diff
@@ -60,24 +25,94 @@ export type NotAutoRestMarkDown = {
...
+  readonly code: "NO_JSON_FILE_FOUND" | "UNREFERENCED_JSON_FILE"
   readonly message: string
   readonly readMeUrl: string
-  readonly openApiUrl: string
+  readonly jsonUrl: string
 }
 
 export type Error = JsonParseError | FileError | NotAutoRestMarkDown
 
+const errorId = (error: Error) => {
```
Comment (sergey-shandar) [inline src/index.ts]: `errorCorrelationId`

File: src/index.ts [hunk 4]
```diff
@@ -60,24 +25,94 @@ export type NotAutoRestMarkDown = {
...
+        return { code: error.code, url: error.readMeUrl }
+      case "JSON_PARSE":
+        return {
+          code: error.code,
+          url: error.error.url,
+          position: error.error.position
+        }
+    }
+  }
+  return JSON.stringify(toObject())
```
Comment (sergey-shandar) [inline src/index.ts]: Because order, it may produce dif results. Use `hash-node-obj` lib?

File: src/index.ts (hunk 1 above)
Comment (sergey-shandar) [reply src/index.ts]: https://github.com/Azure/avocado/issues/16

**Comment by sergey-shandar** [comment]
Examples:
- unmodified specs (no errors): https://dev.azure.com/azure-sdk/public/_build/results?buildId=15510
- altered specs (with unreferenced files): https://dev.azure.com/azure-sdk/public/_build/results?buildId=15517

**Pull request merged**